src/components/DayPicker.jsx

```diff
@@ -242,6 +260,23 @@ class DayPicker extends React.Component {
     return shallowCompare(this, nextProps, nextState);
   }
 
+  componentWillUpdate() {
+    // Calculating the dimensions trigger a DOM repaint which
+    // breaks the CSS transition.
+    // The setTimeout will wait until the transition ends.
+    if (this.calendarInfo) {
+      const { calendarInfoWidth } = this.state;
+      this.timeout = setTimeout(() => {
+        const calendarInfoPanelWidth = calculateDimension(this.calendarInfo, 'width', true, true);
+        if (calendarInfoWidth !== calendarInfoPanelWidth) {
+          this.setState({
+            calendarInfoWidth: calendarInfoPanelWidth,
+          });
+        }
+      }, 200);
```

Comment:
Should we use `transitionDuration` here instead?

Reply:
Makes sense :) I didn't see the props...